src/plugins/vis_types/timeseries/public/convert_to_lens/lib/configurations/xy/layers.ts

```diff
@@ -112,4 +127,77 @@ export const getLayers = (
       };
     }
   });
+  if (!model.annotations || !model.annotations.length) {
+    return nonAnnotationsLayers;
+  }
+
+  const annotationsByIndexPattern = groupBy(
+    model.annotations,
+    (a) => typeof a.index_pattern === 'object' && 'id' in a.index_pattern && a.index_pattern.id
+  );
+
+  const annotationsLayers: Array<XYAnnotationsLayerConfig | undefined> = await Promise.all(
+    Object.entries(annotationsByIndexPattern).map(async ([indexPatternId, annotations]) => {
+      const convertedAnnotations: EventAnnotationConfig[] = [];
+      const { indexPattern } = (await fetchIndexPattern({ id: indexPatternId }, dataViews)) || {};
+
+      if (indexPattern) {
+        annotations.forEach((a: Annotation) => {
+          const lensAnnotation = convertAnnotation(a, indexPattern);
+          if (lensAnnotation) {
+            convertedAnnotations.push(lensAnnotation);
+          }
+        });
+        return {
+          layerId: v4(),
+          layerType: 'annotations',
+          ignoreGlobalFilters: true,
+          annotations: convertedAnnotations,
+          indexPatternId,
+        };
+      }
+    })
+  );
+
+  return nonAnnotationsLayers.concat(...annotationsLayers.filter(nonNullable));
+};
+
+const convertAnnotation = (
+  annotation: Annotation,
+  dataView: DataView
+): EventAnnotationConfig | undefined => {
+  if (annotation.query_string) {
+    const extraFields = annotation.fields
+      ? annotation.fields
+          ?.replace(/\s/g, '')
+          ?.split(',')
+          .map((field) => {
+            const dataViewField = dataView.getFieldByName(field);
+            return dataViewField && dataViewField.aggregatable ? field : undefined;
+          })
+          .filter(nonNullable)
+      : undefined;
+    return {
+      type: 'query',
+      id: annotation.id,
+      label: 'Event',
+      key: {
+        type: 'point_in_time',
+      },
+      color: new Color(transparentize(annotation.color || euiLightVars.euiColorAccent, 1)).hex(),
+      timeField: annotation.time_field,
+      icon:
+        annotation.icon &&
+        ICON_TYPES_MAP[annotation.icon] &&
+        typeof ICON_TYPES_MAP[annotation.icon] === 'string'
```

Comment:
seems like this mapping doesn't fully work:
<img width="1379" alt="Screenshot 2022-09-20 at 11 16 47" src="https://user-images.githubusercontent.com/1508364/191219290-6b05f5e7-3d24-426e-8674-98c35e391baf.png">

Reply:
Thanks Joe, I think the best is to add `starFilled` (as this is the only one that it happens) to Lens as an available option. That was the initial plan when creating the icons selection (to give at least the same options as TSVB) and I somehow omitted it. DO you think it's fine? Commit pushed already.

Reply:
for sure, that seems fine

Reply:
I like stars